test/org/sagebionetworks/bridge/dynamodb/DynamoTaskDaoMockTest.java

```diff
@@ -242,16 +281,17 @@ public void canUpdateTasks() {
         assertEquals(TaskStatus.STARTED, savedTask1.getStatus());
         assertEquals(TaskStatus.FINISHED, savedTask2.getStatus());
     }
-
-    @SuppressWarnings("deprecation")
-    private void assertTask(String dateString, String ref, Task task) {
-        if ("tapTest".equals(dateString)) {
-            assertEquals(ref, task.getActivity().getRef());
-            return;
+    
+    private void assertTask(Task task, String ref, String dateString) {
+        if (task.getActivity().getActivityType() == ActivityType.TASK) {
+            DateTime date = DateTime.parse(dateString);
+            assertTrue(date.isEqual(task.getScheduledOn()));
+            assertEquals(ref, task.getActivity().getTask().getIdentifier());            
+        } else {
+            DateTime date = DateTime.parse(dateString);
+            assertTrue(date.isEqual(task.getScheduledOn()));
+            assertEquals(ref, task.getActivity().getSurvey().getHref());
         }
```

Comment:
Minor: You can refactor the common lines to outside the if/else blocks:
            DateTime date = DateTime.parse(dateString);
            assertTrue(date.isEqual(task.getScheduledOn()));